specs.md

```diff
@@ -19,50 +19,140 @@ This approach is based on data gathered from public automatically resetted netwo
 
 ### Genesis 
 
-With each reset, the network starts again from a new genesis. The new genesis is deterministically created based on the previous one. The main changes in the genesis are chainId, timestamp and the withdrawal credentials of the first validator. 
+To connect to the current instance of the network, client must implement the genesis function. This function defines how the client stores information about the testnet and generates the current genesis. With each reset, network starts from a new genesis which needs to build based on given parameters, and correspond in EL and CL clients.
 
-At the very beginning of testnet existence, in the first iteration, `period 0`, the network starts with `genesis 0`. This very fist genesis is hardcoded in the client like any other predefined network. However, during each client start, client does not initiatilize this genesis but rather builds a new one based on it. This new genesis, derived from the `genesis_0`, will be written to the database and used to run the current network. 
+The main changes in the genesis iteration are chainId, timestamp and the withdrawal credentials of the first validator.
+    
+The network always starts from a genesis which is deterministically created based on the original one, genesis_0. Expiration of the genesis is given by its `genesis_timestamp` and `period` (a predefined constant, length of time a single ephemeral network runs). Therfore once the currenttimestamp reaches the terminal time of ephemeral network, it has to switch to a new genesis. This can be expressed as:
 
-Given a known `period` (the length of time network instance will run), the client can calculate the number of lifecycle iterations from `genesis_0` to the current one and create a new genesis with latest parameters. Always, when the client starts with an option of ephemeral testnet, it checks the current genesis timestamp and identifies whether it's older than `timestamp + period`, at which point it triggers generation of a new genesis.
+    let terminal_timestamp = genesis_timestamp + period
+    
+    if (current_timestamp >= terminal_timestamp):
+       return new_genesis
+    else:
+       return none
```

Comment:
This part should be part of  the reset function described below. Terminal timestamp is not considered in genesis function, only in the reset